libp2p/Host.cpp

```diff
@@ -355,20 +312,42 @@ void Host::runAcceptor()
 				}
 			}
 			
-			if (!success && m_socket->is_open())
+			if (!success && s->is_open())
 			{
 				boost::system::error_code ec;
-				m_socket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
-				m_socket->close();
+				s->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
+				s->close();
 			}
 
 			m_accepting = false;
+			delete s;
```

Comment:
Yes, I meant destructor, and you are right, the documentation says the socket should be in a valid state.
Still, this code could benefit a lot from a bit more RAII. Why do you have to manually call shutdown and close in case of failure, shouldn't a proper destructor do that?